service/src/main/java/com/myhome/services/springdatajpa/UserSDJpaService.java

```diff
@@ -76,6 +89,47 @@
     }).orElse(Optional.empty());
   }
 
+  @Override
+  public boolean requestResetPassword(ForgotPasswordRequest forgotPasswordRequest) {
+    if(forgotPasswordRequest.email != null) {
+      SecurityToken newSecurityToken = securityTokenService.createPasswordResetToken();
+      Optional<User> userOptional = userRepository.findByEmailWithPasswordResetToken(forgotPasswordRequest.email);
+      return userOptional.map(user -> {
+        user.setPasswordResetToken(newSecurityToken);
+        userRepository.save(user);
+        mailService.sendPasswordRecoverCode(user, newSecurityToken.getToken());
+        return true;
+      }).orElse(false);
+    } else {
+      return false;
```

Comment:
We cannot return false from this service which will lead to 400 bad request.
If you do it in this way you create a machine for the attacker to verify if the user exists in the database or now :) 
We can log not successful attempts but we should always return success in case of reset token flow.

Take a look here:
https://stackoverflow.com/a/26404421